src/Workspaces/Core/Portable/LinkedFileDiffMerging/LinkedFileDiffMergingLogger.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Internal.Log;
+using static Microsoft.CodeAnalysis.LinkedFileDiffMergingSession;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal class LinkedFileDiffMergingLogger
+    {
+        private const string Id = nameof(Id);
+        private static ConcurrentDictionary<int, LogAggregator> LogAggregators = new ConcurrentDictionary<int, LogAggregator>();
```

Comment:
Preview workspace is a good point.  I didn't observe many `LogAggregator`s being created during testing, but I'll need to verify.  If we only report telemetry at the end, perhaps we just have a sing `LogAggregator` for all workspaces.